NEWS.md

```diff
@@ -27,6 +27,8 @@
    + Argument `in.place` to `droplevels` has been removed.
    + It's now an error to set `datatable.nomatch`, which has been warning since 1.15.0.
 
+3. The groupingsets() documentation now explicitly addresses scoping issues when using variables from outer environments (e.g., function parameters) in aggregation expressions. Previously, using such variables in j could lead to "object not found" errors due to evaluation environment mismatches. This is resolved by using jj = substitute(expr) instead, which captures both the expression and its environment per R’s scoping rules (Chapter 6 of the R Language Definition manual). Three new examples demonstrate proper usage, including function parameter references (e.g., sum(value > threshold)) and multi-aggregation patterns.
```

Comment:
`substitute()` does not capture the environment (that's the problem that [quosures](https://www.r-project.org/dsc/2017/slides/tidyeval-hygienic-fexprs.pdf) are intended to solve). What it does is _access_ the list or environment given to `substitute()` as an argument (defaults to the caller's environment) at the time of the call. Any symbol that can be directly found in that environment (ignoring the parent chain) is substituted: promises are replaced with the promise expression (losing the promise environment in the process!), other values are substituted directly.